worldguard-bukkit/src/main/java/com/sk89q/worldguard/bukkit/listener/EventAbstractionListener.java

```diff
@@ -357,6 +359,19 @@ public void onEntityChangeBlock(EntityChangeBlockEvent event) {
     @EventHandler(ignoreCancelled = true)
     public void onEntityExplode(EntityExplodeEvent event) {
         Entity entity = event.getEntity();
+        if (entity instanceof WindCharge) {
```

Comment:
I'm not happy with specifying WindCharge at this location. Explosions can have different interaction types:
* KEEP: Not changing anything
* DESTROY, DESTROY_WITH_DELAY: Destroy blocks
* TRIGGER_BLOCK: trigger the blocks.

The current changes check only for WindCharge but not for BreezeWindCharge which also triggers blocks but don't destroy them.

Imo the correct approach should be that Spigot (or Paper) adds the Interaction type of a explosion to EntityExplodeEvent and BlockExplodeEvent. After that we can change our implementation to handle the interaction correctly.

Reply:
> Imo the correct approach should be that Spigot (or Paper) adds the Interaction type of a explosion to EntityExplodeEvent and BlockExplodeEvent. After that we can change our implementation to handle the interaction correctly.

I agree that Spigot or Paper would have to add api to correctly fix this, that check is just a workaround until that api is added. For the specific issue, I could change the check to AbstractWindCharge in the pr, I just inadvertently thought that BreezeWindCharge is a subclass of WindCharge.

Reply:
Created https://hub.spigotmc.org/jira/browse/SPIGOT-7799